addons/purchase/migrations/13.0.1.2/post-migration.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2020 ForgeFlow <http://www.forgeflow.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def move_fields_from_invoice_to_moves(env):
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE account_move_line aml
+        SET purchase_line_id = ail.purchase_line_id
+        FROM account_invoice_line ail
+        WHERE aml.old_invoice_line_id = ail.id"""
+    )
+
+
+def change_type_purchase_order_date_approve(env):
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE purchase_order
+        SET date_approve = {date}::TIMESTAMP AT TIME ZONE 'UTC'
+        WHERE {date} IS NOT NULL
+        """.format(date=openupgrade.get_legacy_name('date_approve'))
+    )
+
+
+def update_account_move_purchase_order_rel_table(env):
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE account_move_purchase_order_rel ampor
+        SET account_move_id = am.id
+        FROM account_invoice ai
+        JOIN account_move am ON am.old_invoice_id = ai.id
+        WHERE ampor.{} = ai.id
+        """.format(openupgrade.get_legacy_name('account_invoice_id'))
+    )
+
+
+def fill_purchase_order_line_qty_received_method(cr):
+    if openupgrade.column_exists(cr, 'purchase_order', 'picking_type_id'):
+        # purchase_stock is installed
+        # set qty_delivered_method = 'stock_moves'
```

Comment:
There is a function to test if a module is installed.

Reply:
Is it better that function than using that column_exists?

Reply:
It is not always possible, because the module can be merged into another